lib/MongoDB/BulkWrite.pm

```diff
@@ -121,8 +121,12 @@ sub _build__client {
     return $self->_database->_client;
 }
 
+# TODO required for sessions...
+sub client { return shift->_client }
+
 with $_ for qw(
   MongoDB::Role::_DeprecationWarner
+  MongoDB::Role::_MaybeClientSession
```

Comment:
I'd rather see session added explicitly as an option to `execute` (or have `execute` get an implicit session from the client).  That's the only place we really need one and minimizes the scope of session usage.  Doing that also eliminates the need for `sub client`.